package/piwebcam/multi-gadget.sh

```diff
@@ -86,14 +86,27 @@ config_usb_webcam () {
 }
 
 # Check if camera is installed correctly
-if [ ! -e /dev/video0 ] ; then
-  echo "I did not detect a camera connected to the Pi. Please check your hardware."
-  CONFIGURE_USB_WEBCAM=false
+if [ -e /dev/video0 ] ; then
+  CONFIGURE_USB_WEBCAM=true
+else
+  COUNT=0
```

Comment:
I suspect all of this delay loop could be simplified by calling 'udevadm settle' before 
" if [ ! -e /dev/video0 ] ; then"

Reply:
That would 100% be a better solution.

Although we're running systemd so perhaps we can go even further... In this case just waiting for https://www.freedesktop.org/software/systemd/man/systemd-udev-settle.service.html to finish should be enough as we assume the camera is attached before the pi is booted, or not at all.

Reply:
ah yes, if there's a systemd way to do this then that's even better indeed.

Reply:
systemd can also wait for a device to appear, both ways are fine.